packages/beacon-node/src/execution/engine/types.ts

```diff
@@ -222,6 +241,32 @@ export interface BlobsBundleRpc {
   proofs: DATA[]; // some ELs could also provide proofs, each 48 bytes
 }
 
+export function getPayloadMethodByFork(fork: ForkName): EngineGetPayloadMethod {
+  switch (true) {
+    case isForkPostElectra(fork):
+      return "engine_getPayloadV4";
+    case isForkBlobs(fork):
+      return "engine_getPayloadV3";
+    case isForkWithdrawals(fork):
+      return "engine_getPayloadV2";
+    default:
+      return "engine_getPayloadV1";
+  }
```

Comment:
I generally prefer `if() return` pattern for such cases. But it's fine if you prefer this way.